pkg/agent/rest.go

```diff
@@ -115,6 +119,26 @@ func FindRendezvouIPAndSSHKeyFromAssetStore(assetDir string) (string, string, er
 	return rendezvousIP, sshKey, nil
 }
 
+// FindAuthTokenFromAssetStore returns the auth token.
+func FindAuthTokenFromAssetStore(assetDir string) (string, error) {
```

Comment:
IIRC we agree to change the approach for the token retrieval, given the fact that the wait-for command could be launched into an environment completely different from the one where the ISO was generated, so fetching it from the asset store doesn't seem a robust option, isn't it?

Reply:
By original design, wait-for always needed the asset store to work correctly. for example, https://github.com/openshift/installer/blob/e9c454889b638a9af3e1203d0585c756f8cf0136/cmd/openshift-install/agent/waitfor.go#L69

Reply:
Had a productive discussion with @andfasano on Slack, where we brainstormed and came up with a technical solution to remove the dependency of the asset store from the `agent wait-for` command. Here are the key points:

1. **Update `FindRendezvouIPAndSSHKeyFromAssetStore`**:
   - Modify this function to read the `rendezvouIP` from an already generated file, instead of relying on the asset store.

2. **Replace SSH Key Check with Ping**:
   - Instead of using the SSH key to test if a node is up and running, we can simply use a ping test to check if the node is reachable.

3. **Persist Auth Token on Disk**:
   - Save the generated auth token on disk, similar to how the `kubeadmin` password is saved.
   - Update `FindAuthTokenFromAssetStore` to read the auth token from this file instead of the state JSON file (asset store), making the `AuthConfig` asset a writable one.


Point 1 and 2 above could be a technical debt for now but we proceed with point 3 with this PR.
cc @andfasano @zaneb @rwsu @bfournie

Reply:
Given that `agent wait-for` has always required the state file and been fine (and metal IPI at least also requires the state file), let's not hold up new work for the sake of rearranging existing stuff.
The auth token is an internal implementation detail and I prefer not to expose it to the user by making it a WritableAsset.
(I _would_ kind of like to get rid of the SSH test and replace it with ping, but this work doesn't need to be blocked on it.)

Reply:
I am ok to proceed with the current implementation and revisit the changes in a new PR

Reply:
I do also agree that removing the already existing binding on rz IP/ssh key retrieved from the state file shouldn't be part of this specific epic - as it looks a (anyhow doable) task not strictly related to it.

By focusing on the auth token only, my previous comment was related to the possibility of creating the image in an environment and monitoring its application in a different one. With the current implementation, IIUC, the user will have to copy as well the `.openshift_installer_state.json` file in the new environment to be able to run the wait-for command. Even though this behavior wasn't an issue until now, it seems to me anyhow a valuable one to be addressed in the future, and in such case I would agree on not blocking now with the current implementation

Reply:
Created a new story https://issues.redhat.com/browse/AGENT-927 to address the above points. @zaneb @andfasano let's continue the discussion on this story about the final approach.